interactive_engine/compiler/src/main/java/com/alibaba/graphscope/gremlin/antlr4/TraversalMethodVisitor.java

```diff
@@ -439,42 +442,52 @@ public Traversal visitTraversalMethod_group_keyby(
             return graphTraversal.by();
         } else if (childCount == 4 && ctx.stringLiteral() != null) {
             return graphTraversal.by(GenericLiteralVisitor.getStringLiteral(ctx.stringLiteral()));
-        } else if (childCount == 4 && ctx.traversalMethod_values() != null) {
-            TraversalMethodVisitor nestedVisitor =
-                    new TraversalMethodVisitor(
-                            gvisitor, GremlinAntlrToJava.getTraversalSupplier().get());
-            Traversal nestedTraversal =
-                    nestedVisitor.visitTraversalMethod_values(ctx.traversalMethod_values());
-            return graphTraversal.by(nestedTraversal);
-        } else if (childCount >= 6 && ctx.traversalMethod_values() != null) {
-            TraversalMethodVisitor nestedVisitor =
-                    new TraversalMethodVisitor(
-                            gvisitor, GremlinAntlrToJava.getTraversalSupplier().get());
-            Traversal nestedTraversal =
-                    nestedVisitor.visitTraversalMethod_values(ctx.traversalMethod_values());
-            if (ctx.traversalMethod_as() != null) {
-                nestedTraversal = nestedVisitor.visitTraversalMethod_as(ctx.traversalMethod_as());
+        } else if (childCount == 4 && ctx.nonStringKeyByList() != null) {
+            GremlinGSParser.NonStringKeyByListContext nonStringCtxs = ctx.nonStringKeyByList();
+            List<Traversal.Admin> keyByList = new ArrayList<>();
+            for (int i = 0; i < nonStringCtxs.getChildCount(); ++i) {
+                GremlinGSParser.NonStringKeyByContext keyByCtx = nonStringCtxs.nonStringKeyBy(i);
+                if (keyByCtx == null) continue;
+                keyByList.add(visitNestedTraversal(keyByCtx.nestedTraversal()).asAdmin());
             }
-            return graphTraversal.by(nestedTraversal);
-        } else if (childCount == 4 && ctx.nestedTraversal() != null) {
-            Traversal nestedTraversal = visitNestedTraversal(ctx.nestedTraversal());
-            return graphTraversal.by(nestedTraversal);
+            ((IrCustomizedTraversal) graphTraversal).by(keyByList);
+            return graphTraversal;
         } else {
             throw new UnsupportedEvalException(
                     ctx.getClass(),
-                    "supported pattern is [group().by('..')] or [group().by(values('..'))] or"
-                            + " [group().by(values('..')).as('..')]");
+                    "supported patterns are [group().by()]"
+                            + " or [group().by('name')] or group().by(non_string_key_list)");
```

Comment:
`non_string_key_list` does not look like a thing that user can understand. Can be more specific

Reply:
These ambiguous error info will be resolved later.